ci/forms.py

```diff
@@ -165,3 +173,40 @@ class ReportGenerationForm(forms.Form):
     email_subject = forms.CharField(
         widget=forms.TextInput(attrs={'placeholder': 'Subject for email'}),
         required=False)
+
+
+class MsisdnReportGenerationForm(ReportGenerationForm):
+    msisdn_list = forms.FileField()
+
+    def clean_msisdn_list(self):
+        msisdns = []
+        wb = load_workbook(self.cleaned_data['msisdn_list'])
+        ws = wb[wb.sheetnames[0]]  # There should only be one sheet
+        # Get msisdn column
+        msisdn_column = None
+        for i, column in enumerate(ws.columns):
+            if column[0].value is None:
+                # Skip untitled columns
+                continue
+            value = str(column[0].value)
+            if value.lower() == 'phone number':
+                msisdn_column = i
+                break
+        if msisdn_column is None:
+            raise forms.ValidationError(
+                "Invalid contents for: %(file)s. File must contain "
+                "'Phone number' column",
+                code='invalid',
+                params={'file': self.cleaned_data['msisdn_list']})
+
+        for row in ws.rows:
+            value = row[msisdn_column].value
+            if value is None:
+                # Skip empty Rows
+                continue
+            value = str(value)
+            if value.lower() == 'phone number':
+                # Skip the header row
+                continue
+            msisdns.append(value)
```

Comment:
Do we need to do anything to normalise these msisdns?

Reply:
@rudigiesler I was thinking of checking that they were at least all numerical but I got lazy. I'll add it. Anything more than that I'd say should be in the registration app because things like country code and length are project dependent

Reply:
I'm not sure that we want them to be numerical, eg `+27741234567` contains a `+` character that we actually want there. Maybe we should just do normalisation in the registration app that knows about these things.

Reply:
@rudigiesler the commit I just added removes the `+` before checking if the string is a digit. It then adds the `+` if it wasn't there before